tests/diffDataParent.js

```diff
@@ -0,0 +1,196 @@
+            assert.isUndefined(text1.p.dataParent);
+        });
+
+        it("should have proper data parent after reloading the form", function() {
+            util.loadXML("");
+            var text1 = util.addQuestion("Text", 'text1'),
+                form = call("getData").core.form;
+
+            util.addQuestion("Group", 'group1');
+
+            text1.p.dataParent = '/data/group1';
+            util.loadXML(call("createXML"));
+            assert.equal(text1.p.dataParent, '/data/group1');
+            util.assertTreeState(form.dataTree(),
+                "group1",
+                "  text1"
+            );
+        });
+
+        it("should parse and write XML to have the same order", function() {
+            util.loadXML(PARSE_XML);
+            util.assertXmlEqual(call("createXML"), PARSE_XML);
+        });
+
+        it("should have proper data parent after being loaded from xml", function() {
+            util.loadXML(PARSE_XML);
+            var question1 = util.getMug("/data/question4/question1");
+            assert.equal(question1.p.dataParent, "/data/question4");
+            util.assertJSTreeState(
+                "question1",
+                "question4",
+                "  question3"
+            );
+        });
+
+        it("should use the control parent when inserting after a mug with different data parent", function() {
+            util.loadXML("");
+            var form = call("getData").core.form,
+                text1 = util.addQuestion("Text", 'text1');
+            util.addQuestion("Group", 'group');
+            util.addQuestion("Text", 'text2');
+
+            text1.p.dataParent = '/data/group';
+            util.addQuestion.bind({prevId: text1.p.nodeID})("Text", 'text3');
+            // questions with alternate dataParent always come last in the data tree
+            util.assertTreeState(form.dataTree(),
+                "text3",
+                "group",
+                "  text2",
+                "  text1"
+            );
+            util.assertJSTreeState(
+                "text1",
+                "text3",
+                "group",
+                "  text2"
+            );
+        });
+
+        it("should properly load group before sibling/child", function() {
+            util.loadXML(SIBLING_AS_CHILD);
+            var text = util.getMug("/data/text");
+            assert.equal(text.p.dataParent, "/data");
+            util.assertJSTreeState(
+                "group",
+                "  text"
+            );
+            util.assertXmlEqual(call("createXML"), SIBLING_AS_CHILD);
+        });
+
+        it("should properly load mug after other mugs", function() {
+            util.loadXML(DATA_PARENT_MUG_AFTER);
+            var text = util.getMug("/data/question4/question1");
+            assert.equal(text.p.dataParent, "/data/question4");
+            util.assertJSTreeState(
+                "question3",
+                "question1",
+                "question4"
```

Comment:
This PR is looking good. One final nitpick. The question names in these tests are rather incomprehensible at first glance. What do you think of using names like
```js
"before-child",
"child",
"parent"
```
Also might want to rename the local `text` variable to `child`.